### PR TITLE
asserts: implement snap-resource-revision

### PR DESCRIPTION
**Pull request by pedronis**
including delegation support parallel to snap-revision similar support

### CODE REVIEW

**Comment by suligap** [comment]
Looks good to me, thanks.

File: asserts/snap_resource_asserts.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015-2022 Canonical Ltd
```
Comment (alfonsosanchezbeato) [inline asserts/snap_resource_asserts.go:4]: nitpick: should be 2023

File: asserts/snap_resource_asserts.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+			return fmt.Errorf("snap-resource-revision assertion with provenance %q for snap id %q is not signed by an authorized authority: %s", resrev.Provenance(), resrev.SnapID(), resrev.AuthorityID())
+		}
+	}
+	return nil
+}
+
+// expected interface is implemented
+var _ consistencyChecker = (*SnapResourceRevision)(nil)
+
+// Prerequisites returns references to this snap-resourcerevision's prerequisite assertions.
```
Comment (alfonsosanchezbeato) [inline asserts/snap_resource_asserts.go:134]: s/snap-resourcerevision/snap-resource-revision/

File: asserts/snap_resource_asserts_test.go [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+ *
+ */
+
+package asserts_test
+
+import (
+	//"encoding/base64"
+	//"os"
+	//"path/filepath"
+	//"sort"
```
Comment (alfonsosanchezbeato) [inline asserts/snap_resource_asserts_test.go:26]: All this packages commented out can be removed

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM (just some minor leftovers/typos that I've seen)

**Review by MiguelPires — APPROVED** [review]
lgtm

**Pull request merged**